arrows/ocv/resection_camera.cxx

```diff
@@ -99,9 +132,22 @@ ::check_configuration( vital::config_block_sptr config ) const
     LOG_ERROR( d_->m_logger,
                "max iterations is " << max_iterations <<
                ", needs to be greater than zero." );
-    good_conf = false;
+    return false;
+  }
+
+  std::stringstream ss( config->get_value< std::string > ( "focal_scales", "1" ) );
+  vectorf focal_scales;
+  ss >> focal_scales;
+  auto m = std::min_element( focal_scales.begin(), focal_scales.end() );
+  if( m == focal_scales.end() || *m <= 0 )
```

Comment:
I think `focal_scales` being empty should generate a different error message than the error for values less than zero.

I'm also wondering if we should require that "1" be in the list.  Or maybe this should be a list of _additional_ focal scales and "1" is just implicitly assumed to be always in the list.  In that case if the list was empty by default you would just get "1" and if the list was "0.5 2" you would end up using scales of 0.5, 1, and 2.

Reply:
We now ensure that focal_scales is never empty, by default containing 1.0, but allowing custom lists specified in a configuration, including the case omitting 1.0 for some special cases, when 1.0 is not of interest, thus achieving the most flexibility.